homeassistant/components/hassio/__init__.py

```diff
@@ -196,7 +196,7 @@ def valid_addon(value: Any) -> str:
     {
         vol.Optional(ATTR_HOMEASSISTANT): cv.boolean,
         vol.Optional(ATTR_FOLDERS): vol.All(cv.ensure_list, [cv.string]),
-        vol.Optional(ATTR_ADDONS): vol.All(cv.ensure_list, [cv.slug]),
```

Comment:
Why not updating the slug validator ? 

https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/config_validation.py#L574-L582

Reply:
The goal here is to allow what Supervisor allows as slug for add-ons. Add-ons learn their slug from their YAML configuration file, so it is user supplied, but validated with the following regex: `[-_.A-Za-z0-9]+` (see https://github.com/home-assistant/supervisor/blob/2024.04.1/supervisor/addons/const.py#L48).

Changing that would be a major breaking change, render many add-on configuration incorrect. We can't really do that.

So what I aim to do here, is make Core pass any slug that the Supervisor allows for add-ons. That is currently not the case, and hence folks can't take backups using service calls for slugs with dashes (see https://github.com/home-assistant/core/issues/107693).

We can't really change the current slugify validator, as it is used in other places. If anything, we'd need to extend (or write a new) slugify validator to support this type of slugs.

I've investigated into it, however, it doesn't seem that trivial: The slug validator is based on the slugify package itself. From a quick investigation, the way slugify currently works, it always replaces hyphens with whatever is considered the default separator (see https://github.com/un33k/python-slugify/blob/v8.0.4/slugify/slugify.py#L194-L195).. Allowing two separator types (hyphens and dashes) would really need different slugify code :cry:

Reply:
Looking at the code again, I actually realized there is already an existing regex validator. It seems I can simply use that to validate the slug correctly :tada: 

I've updated the PR. Thanks for your review!

Reply:
I thought the same.
Glad I was able to help a bit